src/screenshotdialog.cpp

```diff
@@ -157,7 +157,7 @@ void ScreenshotDialog::doScreenshot() {
         }
         else { // XFixes returns long integers which is not 32 bit on 64 bit systems.
           long len = cursor->width * cursor->height;
-          quint32* buf = new quint32[len];
+          buf = new quint32[len];
           for(long i = 0; i < len; ++i)
             buf[i] = (quint32)cursor->pixels[i];
           cursorImage = QImage((uchar*)buf, cursor->width, cursor->height, QImage::Format_ARGB32);
```

Comment:
I think it's better to delete [] buf within this block, and remove the 'buf' declaration from line 153.